docs/DeveloperGuide.md

```diff
@@ -209,9 +209,50 @@ The following activity diagram summarizes what happens when a user executes a `v
   * Cons: List of clients will be replaced by the Client details, extra effort to list Clients again.
 
 
-### Check in/ out feature
+### Check in/out feature
 
-_{Explain here how the check in/out feature will be implemented}_
+#### Implementation
+
+The **Check Client** mechanism is facilitated by `CheckClientCommand` extending the `Command` family of classes, as elaborated in the Logic component of MATER. Additionally, it requires the following operations:
+
+* `CheckClientCommandParser#parse()` - Given the argument succeeding the `check-client` command, parses the appropriate index to `CheckClientCommand`.
+* `CheckClientCommand#execute()` - Given the list of Clients, identifies the indexed Client and toggles the "checked (in or out)" status of the Client's car.
+
+Given below is an example usage scenario and details on how the **Check Client** mechanism behaves at each step.
+
+**Step 1**: The user launches the application, where all clients are listed by default.
+
+**Step 2**: The user executes `check-client 1` command to toggle the "checked-in" status of the car for the first Client in the list.
```

Comment:
I think should have more consistency, above is "checked (in or out)" status, here is only "checked-in" status.